src/paradigma/heart_rate/heart_rate_analysis.py

```diff
@@ -64,25 +65,84 @@ def signal_quality_classification(df: pd.DataFrame, config: SignalQualityClassif
 
     # Prepare the data
     lr_clf.feature_names_in_ = ['var', 'mean', 'median', 'kurtosis', 'skewness', 'f_dom', 'rel_power', 'spectral_entropy', 'signal_to_noise', 'auto_corr']
-    X = df.loc[:, lr_clf.feature_names_in_]
+    X = df_windowed.loc[:, lr_clf.feature_names_in_]
 
     # Normalize features using mu and sigma
     X_normalized = X.copy()
     for idx, feature in enumerate(lr_clf.feature_names_in_):
         X_normalized[feature] = (X[feature] - mu[idx]) / sigma[idx]
 
     # Make predictions for PPG signal quality assessment
-    df[DataColumns.PRED_SQA_PROBA] = lr_clf.predict_proba(X_normalized)[:, 0]                   
-    df.drop(columns = lr_clf.feature_names_in_, inplace=True)  # Drop the features used for classification since they are no longer needed
-    
-    return df    
+    df_sqa = df_windowed.copy()
+    df_sqa[DataColumns.PRED_SQA_PROBA] = lr_clf.predict_proba(X_normalized)[:, 0]                   
+    df_sqa.drop(columns = lr_clf.feature_names_in_, inplace=True)  # Drop the features used for classification since they are no longer needed
```

Comment:
New best practices omit using `inplace` and instead recommend using direct object assignment. Again, not necessary per se, just a recommendation.

Reply:
Agree, I updated this and chained the operation of copying, assigning and dropping in one to keep it concise, clear and functional